### PR TITLE
Added size argument

### PR DESCRIPTION
**Pull request by Dam-Buty**
Hi! I loved your script so i just added a --size argument (-s) to set the pixel_size of obscure_rectangle. It's dead simple and just checks if the argument is an integer before passing it to the functions. Default is 9, as was in your code.

I haven't coded in C since forever so if there is any coding convention i should know about please let me know and i'll rework it.

### CODE REVIEW

**Comment by Dam-Buty** [comment]
While i was at it i added a --fuzzy option that applies a blur to the image. It is relatively fast because it blurs the reduced version of the image before it is re-expanded.

File: xwobf.c [hunk 1]
```diff
@@ -43,22 +43,38 @@ size_t        rect_size = 0;
 void print_usage()
 {
     printf("Usage: xwobf [OPTION]... DEST\n");
-    printf("  -h --help\tprint this message and exit\n");
+    printf("  -h --help\t\tprint this message and exit\n");
+    printf("  -s pixel_size\t\tadjust the obfuscation strength (default=9)\n");
+    printf("  --size=pixel_size\n");
```
Comment (glindstedt) [inline xwobf.c]: trailing whitespace

File: xwobf.c [hunk 2]
```diff
@@ -43,22 +43,38 @@ size_t        rect_size = 0;
...
         {NULL, no_argument, NULL, 0}
     };
 
     int c;
     while ((c = getopt_long(argc, argv, optstring, longopts, NULL)) != -1) {
         switch(c) {
+            case 's':
+                if (sscanf (optarg, "%i", &pixel_size)!=1) {
+                    puts("Size argument should be an integer");
+                    exit(EXIT_FAILURE);
```
Comment (glindstedt) [inline xwobf.c]: trailing whitespace

File: xwobf.c [hunk 3]
```diff
@@ -156,23 +172,25 @@ void cleanup()
...
     if ((obs_wand = CloneMagickWand(wand))) {
         (void)MagickCropImage(obs_wand, rec->w, rec->h, rec->x, rec->y);
 
         // This is where the magick happens
-        size_t pixel_size = 9;
         (void)MagickResizeImage(obs_wand, (rec->w)/pixel_size, (rec->h)/pixel_size,
                 PointFilter, 0);
+        if (fuzzy) {
+                (void)MagickBlurImage(obs_wand, 0, 1);
+        }
```
Comment (glindstedt) [inline xwobf.c]: trailing whitespace

**Comment by Dam-Buty** [comment]
Is there some script that warns you of these things or do you just click on every line to see whitespace?

**Comment by glindstedt** [comment]
Thanks for the pull request @Dam-Buty! Looks good overall, although I found one bug. Other than that I just have a few little nitpicks :)

If you try to run it with 0 as size it will crash due to a division by zero on line [189](https://github.com/Dam-Buty/xwobf/blob/master/xwobf.c#L189). I think the best way to solve this at the moment is to make `obscure_rectangle` treat anything < 1 as 1, since negative integers doesn't make sense either.

Also there were a few lines with trailing whitespace, which I marked out. Also could you make sure that the functions have the same name for the `blur_size`/`pixel_size` parameter? Most notably `obscure_image` and `obscure_rectangle` have different names for the parameter in `xwobf.c` but they both have `blur_size` as the name in `xwobf.h`. Pick whichever name you think fits best, consistency is the key :)

**Comment by glindstedt** [comment]
I pulled the branch and my editor gave warnings about the lines with trailing whitespace. Unfortunately they're invisible on github.

**Comment by Dam-Buty** [comment]
Hi @glindste , thanks for the nitpicking! I definitely could use some more attention to detail...

I corrected what you mentioned, and treated the size < 1 case directly in the main function with a message to the user. It seemed to me that it was a user input problem, and that there was no point in propagating an unusable value.

**Comment by glindstedt** [comment]
Yeah it's probably better checking the input before it even gets there so we don't add extra complexity to the function. Anyway, looks good to me! Thank you for your contribution @Dam-Buty :)

**Pull request merged**